src/node_buffer.cc

```diff
@@ -1051,6 +1054,75 @@ static void EncodeUtf8String(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+static void EncodeInto(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+  Local<Context> context = env->context();
+  CHECK_GE(args.Length(), 2);
+  CHECK(args[0]->IsString());
+  CHECK(args[1]->IsUint8Array());
+
+  size_t read = 0;
+  size_t written = 0;
+
+  Utf8Value src(isolate, args[0]);
+  icu::UnicodeString utf16 = icu::UnicodeString::fromUTF8(
```

Comment:
Doesn't this now creates 2 copies of the original string?

I still feel that if we can't guarantee that our implementation `encodeInto` does not create copies of the original string, we should either do not support it at all, or note that in the docs (even though it is a bit weird to say that we creates copies for an API that is designed to avoid that copy).

Reply:
@joyeecheung  yes, it does.

I implement a new without copy the original string.  I am looking forward your feedback, thank you.